### PR TITLE
Module REST task implementation in the new feature branch

### PR DESCRIPTION
**Pull request by vakser**
Implement task in module REST

### CODE REVIEW

File: src/main/java/com/epam/learn/springcore/controller/TraineeController.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                    content = @Content),
+            @ApiResponse(responseCode = "404", description = "Trainee not found", content = @Content),
+            @ApiResponse(responseCode = "401", description = "Request lacks valid authentication credentials", content = @Content)
+    })
+    @PatchMapping("/{username}")
+    public ResponseEntity<Void> changeActivationStatus(@PathVariable String username,
+                                                       @Valid @RequestBody ActivationRequest activationRequest,
+                                                       @RequestHeader HttpHeaders headers) {
+        if (headers.containsKey(HttpHeaders.AUTHORIZATION)) {
+            String token = Objects.requireNonNull(headers.get(HttpHeaders.AUTHORIZATION)).get(0);
```
Comment (Jossindev) [inline src/main/java/com/epam/learn/springcore/controller/TraineeController.java]: You can move this logic to facade layer

File: src/main/java/com/epam/learn/springcore/dao/TrainingRepository.java [hunk 2]
```diff
@@ -1,14 +1,23 @@
...
 
 import java.util.List;
 
-public interface TrainingRepository extends JpaRepository<Training, Integer> {
-//    List<Trainer> findUnassignedTrainers(String traineeUsername);
-//    void updateTrainersListForTrainee(String traineeUsername, List<Trainer> trainers);
-    Training findByTraineeAndTrainer(Trainee trainee, Trainer trainer);
+public interface TrainingRepository extends JpaRepository<Training, Integer>, JpaSpecificationExecutor<Training> {
+
+    default List<Training> findByCriteria(TraineeTrainingSearchCriteria criteria) {
```
Comment (Jossindev) [inline src/main/java/com/epam/learn/springcore/dao/TrainingRepository.java]: why did you use default?

Comment (Jossindev) [reply src/main/java/com/epam/learn/springcore/dao/TrainingRepository.java]: It is not good to send one entity and then move it to another inside repository method

File: src/main/java/com/epam/learn/springcore/controller/TraineeController.java (hunk 1 above)
Comment (vakser) [reply src/main/java/com/epam/learn/springcore/controller/TraineeController.java]: Moved

File: src/main/java/com/epam/learn/springcore/dao/TrainingRepository.java (hunk 2 above)
Comment (vakser) [reply src/main/java/com/epam/learn/springcore/dao/TrainingRepository.java]: Updated

**Pull request merged**